packages/runtime-core/src/apiDefineComponent.ts

```diff
@@ -5,22 +5,72 @@ import {
   ComponentOptionsWithArrayProps,
   ComponentOptionsWithObjectProps,
   ComponentOptionsMixin,
-  RenderFunction
+  RenderFunction,
+  ComponentOptionsBase
 } from './componentOptions'
 import {
   SetupContext,
-  FunctionalComponent,
   AllowedComponentProps,
   ComponentCustomProps
 } from './component'
-import {
-  CreateComponentPublicInstance,
-  ComponentPublicInstanceConstructor
-} from './componentProxy'
 import { ExtractPropTypes, ComponentPropsOptions } from './componentProps'
 import { EmitsOptions } from './componentEmits'
 import { isFunction } from '@vue/shared'
 import { VNodeProps } from './vnode'
+import {
+  CreateComponentPublicInstance,
+  ComponentPublicInstance
+} from './componentProxy'
+
+declare const JSX: unique symbol
```

Comment:
What is this for?

Reply:
I started by moving the JSX constructor to jsx.d.ts file, but because of our h overloads tights to the constructor syntax, didn't make much sense to have in other file, since it didn't work without the jsx.d.ts to have been imported by the typescript.